File: app/src/androidTest/java/it/droidcon/testingdaggerrxjava/test1/TestStackOverflowServiceModule.java
package it.droidcon.testingdaggerrxjava.test1;

import dagger.Module;
import dagger.Provides;
import it.droidcon.testingdaggerrxjava.core.gson.StackOverflowService;
import javax.inject.Singleton;
import org.mockito.Mockito;

@Module
public class TestStackOverflowServiceModule {
    @Provides @Singleton public StackOverflowService provideStackOverflowService() {
        return Mockito.mock(StackOverflowService.class);
    }
}

File: app/src/androidTest/java/it/droidcon/testingdaggerrxjava/test1/TestStackOverflowServiceModule.kt
package it.droidcon.testingdaggerrxjava.test1

import dagger.Module
import dagger.Provides
import it.droidcon.testingdaggerrxjava.core.gson.StackOverflowService
import org.mockito.Mockito
import javax.inject.Singleton

@Module class TestStackOverflowServiceModule {
    @Provides @Singleton fun provideStackOverflowService(): StackOverflowService =
            Mockito.mock(StackOverflowService::class.java)
}
